packages/flame/lib/src/components/component_set.dart

```diff
@@ -71,29 +72,45 @@ class ComponentSet extends QueryableOrderedSet<Component> {
   /// You can await for the onLoad function, if present.
   /// This method can be considered sync for all intents and purposes if no
   /// onLoad is provided by the component.
-  Future<void> addChild(Component c, {BaseGame? gameRef}) async {
-    prepare(c, gameRef: gameRef);
-
-    final loadFuture = c.onLoad();
-    if (loadFuture != null) {
-      await loadFuture;
+  Future<void> addChild(Component component) async {
+    print('Trying to addChild $component to $this');
```

Comment:
print

Reply:
There's a linter rule https://dart-lang.github.io/linter/lints/avoid_print.html to help catch these

Reply:
Since this is a draft I'm having these to debug things, I'll remove them once I mark it as "Ready for review". :)

Reply:
There is https://dart-lang.github.io/linter/lints/avoid_print.html linter rule to catch these kinds of things